R/ml_decision_tree.R

```diff
@@ -57,25 +79,48 @@ ml_decision_tree <- function(x,
   schema <- sdf_schema(df)
   responseType <- schema[[response]]$type
 
-  regressor  <- "org.apache.spark.ml.regression.DecisionTreeRegressor"
-  classifier <- "org.apache.spark.ml.classification.DecisionTreeClassifier"
-
-  envir$model <- if (identical(type, "regression"))
-    regressor
-  else if (identical(type, "classification"))
-    classifier
-  else if (responseType %in% c("DoubleType", "IntegerType"))
-    regressor
-  else
-    classifier
-
-  rf <- invoke_new(sc, envir$model)
-
-  model <- rf %>%
+  modelType <- if (identical(type, "regression"))
```

Comment:
Yep. I also think we should have helper functions for most of the common inputs so we're not copy-pasting a bunch of `ensure_scalar_`-`if-stop` code. Thought we'd do it in another PR though.